File: dsa401/Recursion Questions/DigitalSum.java
public class DigitalSum {
    static int SumDigital(int n) {
        if (n / 10 == 0) {
            return n;
        } else {
            return n % 10 + SumDigital(n / 10);
        }
    }

    public static void main(String[] args) {
        int x = SumDigital(12345);
        if (x > 9) {
            int y = SumDigital(x);
            System.out.println(y);
        } else {
            System.out.println(x);
        }
    }
}

File: dsa401/Recursion Questions/DigitalSum.js
function DigitalSum(num) {
  if (num / 10 == 0) {
    return num;
  } else {
    return (num % 10) + DigitalSum(Math.floor(num / 10));
  }
}
let ans = DigitalSum(12345);
if (ans > 9) {
  let z = DigitalSum(ans);
  console.log(z);
} else {
  console.log(ans);
}
